src/api/workspace/workspaceRouter.ts

```diff
@@ -0,0 +1,64 @@
+import { OpenAPIRegistry } from '@asteasolutions/zod-to-openapi';
+import express, { Router } from 'express';
+
+import { CreateWorkspaceSchema, GetWorkspaceSchema, WorkspaceSchema } from '@/api/workspace/workspaceModel';
+import { createApiResponse } from '@/api-docs/openAPIResponseBuilders';
+import { validateRequest } from '@/common/utils/httpHandlers';
+
+import WorkspaceController from './workspaceController';
+
+export const workspaceRegistry = new OpenAPIRegistry();
+
+const bearerAuth = workspaceRegistry.registerComponent('securitySchemes', 'bearerAuth', {
+	type: 'http',
+	scheme: 'bearer',
+	bearerFormat: 'JWT',
+});
+
+workspaceRegistry.register('Workspace', WorkspaceSchema);
+
+export const workspaceRouter: Router = (() => {
+	const router = express.Router();
+
+	workspaceRegistry.registerPath({
+		method: 'post',
+		path: '/workspaces/CreateWorkspace',
+		tags: ['Workspace'],
+		security: [{ [bearerAuth.name]: [] }],
+		request: {
+			body: {
+				content: {
+					'application/json': {
+						schema: CreateWorkspaceSchema.shape.body,
+					},
+				},
+			},
+		},
+		responses: createApiResponse(WorkspaceSchema, 'Success'),
+	});
+
+	router.post('/CreateWorkspace', [validateRequest(CreateWorkspaceSchema)], WorkspaceController.createWorkspace);
+
+	workspaceRegistry.registerPath({
+		method: 'get',
+		path: '/workspaces/GetAllWorkspaces',
+		tags: ['Workspace'],
+		security: [{ [bearerAuth.name]: [] }],
+		request: {},
+		responses: createApiResponse(WorkspaceSchema, 'Success'),
+	});
+
+	router.get('/GetAllWorkspaces', WorkspaceController.getWorkspaces);
+	workspaceRegistry.registerPath({
+		method: 'get',
+		path: '/workspaces/{id}',
+		tags: ['Workspace'],
+		security: [{ [bearerAuth.name]: [] }],
+		request: { params: GetWorkspaceSchema.shape.params },
+		responses: createApiResponse(WorkspaceSchema, 'Success'),
+	});
+
+	router.get('/:id', validateRequest(GetWorkspaceSchema), WorkspaceController.getWorkspaceById);
+
+	return router;
+})();
```

Comment:
Consider adding an **authentication** to the middleware array and create an **authorization** middleware, I tried creating a `workspace` with a different `ownerId` than mine and it worked, the same goes for creating a `channel`. 

I also expect that users outside the `workspace` or `channel` can't retrieve data about the channel, you can use the `members` table to check for access and create a new table representing the `many to many` relationship between users and workspaces, which you can use later as a reference for the authorization step.


```
{
  "success": false,
  "message": "Error creating Workspace: insert into `workspaces` (`avatarUrl`, `description`, `name`, `ownerId`) values ('string', 'string', 'string', 1) - Cannot add or update a child row: a foreign key constraint fails (`wavely`.`workspaces`, CONSTRAINT `workspaces_ownerid_foreign` FOREIGN KEY (`ownerId`) REFERENCES `users` (`id`))",
  "responseObject": null,
  "statusCode": 500
}
```